test-project/Assets/EditmodeTests/GameObjectRepresentation/MonoBehaviourActivationManagerTests.cs

```diff
@@ -0,0 +1,347 @@
+using Generated.Improbable;
+using Generated.Improbable.Gdk.Tests.ComponentsWithNoFields;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.Commands;
+using Improbable.Worker.Core;
+using NUnit.Framework;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.GameObjectRepresentation.EditModeTests.MonoBehaviourActivationManagerTests
+{
+    public abstract class ActivationManagerTestBase<TBehaviour> where TBehaviour : Component
+    {
+        private World world;
+        protected GameObject TestGameObject;
+        private EntityManager entityManager;
+        protected MonoBehaviourActivationManager ActivationManager;
+
+        [SetUp]
+        public void SetUp()
+        {
+            world = new World("test-world");
+            entityManager = world.GetOrCreateManager<EntityManager>();
+            TestGameObject = new GameObject();
+
+            TestGameObject.AddComponent<TBehaviour>();
+            TestGameObject.AddComponent<SpatialOSComponent>().Entity = entityManager.CreateEntity();
+
+            var loggingDispatcher = new LoggingDispatcher();
+            var injectableStore = new InjectableStore();
+            var requiredFieldInjector = new RequiredFieldInjector(entityManager, loggingDispatcher);
+
+            ActivationManager = new MonoBehaviourActivationManager(TestGameObject,
+                requiredFieldInjector, injectableStore, loggingDispatcher);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            Object.DestroyImmediate(TestGameObject);
```

Comment:
Nit: et's use the `UnityObjectDestroyer` across the board for destroying things. :)